core/drivers/crypto/versal/ecc.c

```diff
@@ -207,7 +229,15 @@ static TEE_Result verify(uint32_t algo, struct ecc_public_key *key,
 	memset(&arg, 0, sizeof(arg));
 
 	/* Verify the message with the validated key */
-	versal_mbox_alloc(sig_len, sig, &s);
+	swp = calloc(1, sig_len);
+	if (!swp) {
+		ret = TEE_ERROR_OUT_OF_MEMORY;
+		goto out;
+	}
+	eswap(sig_len / 2, sig, swp);
```

Comment:
Check at function entry that `sig_len` is even?

Reply:
ECDSA signatures are 2 times the key length but I'll check anyway,